service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java

```diff
@@ -0,0 +1,374 @@
+
+    @Test
+    public void updateMushroomHuntersInfo() throws Exception {
+        String newUpdateValue = "New Info";
+        MushroomHunter hunter = service.findHunterById(persistedHunter.getId());
+        assertThat(hunter.getPersonalInfo()).isNotEqualTo(newUpdateValue);
+
+        hunter.setPersonalInfo(newUpdateValue);
+        service.updateHunter(hunter);
+
+        hunter = service.findHunterById(persistedHunter.getId());
+        assertThat(hunter.getPersonalInfo()).isEqualTo(newUpdateValue);
+
+    }
+
+    @Test
+    public void updateMushroomHuntersNicknameToNull() throws Exception {
+        String newUpdateValue = null;
+        MushroomHunter hunter = service.findHunterById(persistedHunter.getId());
+        hunter.setUserNickname(newUpdateValue);
+
+        assertThatThrownBy(() -> service.updateHunter(hunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void updateMushroomHuntersToNicknameDuplicate() throws Exception {
+        String newUpdateValue = persistedHunter2.getUserNickname();
+        MushroomHunter hunter = service.findHunterById(persistedHunter.getId());
+        hunter.setUserNickname(newUpdateValue);
+
+        assertThatThrownBy(() -> service.updateHunter(hunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void updateMushroomHuntersFirstNameToNull() throws Exception {
+        String newUpdateValue = null;
+        MushroomHunter hunter = service.findHunterById(persistedHunter.getId());
+        hunter.setFirstName(newUpdateValue);
+
+        assertThatThrownBy(() -> service.updateHunter(hunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void updateMushroomHuntersSurnameToNull() throws Exception {
+        String newUpdateValue = null;
+        MushroomHunter hunter = service.findHunterById(persistedHunter.getId());
+        hunter.setSurname(newUpdateValue);
+
+        assertThatThrownBy(() -> service.updateHunter(hunter)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+
+    @Test
+    @Ignore
+    public void updatePassword() throws Exception {
+        assertThat(service.authenticate(persistedHunter, "winry")).isTrue();
+        service.updatePassword(persistedHunter, "winry", "hudsonsoft");
+        assertThat(service.authenticate(persistedHunter, "winry")).isFalse();
+        assertThat(service.authenticate(persistedHunter, "hudsonsoft")).isTrue();
+    }
+
+    @Test
+    @Ignore
+    public void updatePasswordToNull() throws Exception {
+        assertThatThrownBy(() -> service.updatePassword(persistedHunter, "winry", null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    @Ignore
+    public void authenticate() throws Exception {
+        assertThat(service.authenticate(persistedHunter, "winry")).isTrue();
+    }
+
+    @Test
+    public void authenticateWithNullPassword() throws Exception {
+        assertThat(service.authenticate(persistedHunter, null)).isFalse();
```

Comment:
shouldn't this throw an exception?

Reply:
not in the way it is implemented:

 public static boolean validatePassword(String password, String correctHash) {
        if (password == null) return false;